As of Python 3.9+ typing.List is deprecated. For more information che…

bcatalfo opened the pull request:
…ck https://docs.python.org/3/library/typing.html#typing.List

bcatalfo committed:
As of Python 3.9+ typing.List is deprecated. For more information check https://docs.python.org/3/library/typing.html#typing.List

bcatalfo committed:
General: I use black (python linter) so a variety of automatic changes
such as changing single quotes to double quotes
BubbleSort: Removed unnecessary comments
HeapSort: Skip calling heapify on leaf nodes (wastes n/2 operations)
InsertionSort: Changed comments
MergeSort: define aux inline (better practice)
SelectionSort: Using TypeVar gives an error in linters as you only
reference T once, so I changed it to Any

bcatalfo committed:
added helpful comment

pep8speaks commented:
Hello @YeezyYeezyWhatsGood! Thanks for updating this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:



* In the file [`Python/HeapSort.py`](https://github.com/diptangsu/Sorting-Algorithms/blob/47cdfabe4d36c7dced5cc0d414d29c3f10767d3e/Python/HeapSort.py):

> [Line 5:80](https://github.com/diptangsu/Sorting-Algorithms/blob/47cdfabe4d36c7dced5cc0d414d29c3f10767d3e/Python/HeapSort.py#L5): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (96 > 79 characters)
> [Line 19:80](https://github.com/diptangsu/Sorting-Algorithms/blob/47cdfabe4d36c7dced5cc0d414d29c3f10767d3e/Python/HeapSort.py#L19): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (88 > 79 characters)
> [Line 26:80](https://github.com/diptangsu/Sorting-Algorithms/blob/47cdfabe4d36c7dced5cc0d414d29c3f10767d3e/Python/HeapSort.py#L26): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (83 > 79 characters)
> [Line 29:80](https://github.com/diptangsu/Sorting-Algorithms/blob/47cdfabe4d36c7dced5cc0d414d29c3f10767d3e/Python/HeapSort.py#L29): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (97 > 79 characters)

* In the file [`Python/InsertionSort.py`](https://github.com/diptangsu/Sorting-Algorithms/blob/47cdfabe4d36c7dced5cc0d414d29c3f10767d3e/Python/InsertionSort.py):

> [Line 8:80](https://github.com/diptangsu/Sorting-Algorithms/blob/47cdfabe4d36c7dced5cc0d414d29c3f10767d3e/Python/InsertionSort.py#L8): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (83 > 79 characters)









##### Comment last updated at 2022-08-05 04:45:56 UTC

bcatalfo committed:
It's more pythonic to handle the edge case first then proceed with the
base case- by doing so we use less indentation and get more readable
code. Remember "Flat is better than nested" - The Zen of Python
(type import this into a python interpreter)

bcatalfo committed:
In Python the start of a range defaults to 0 so by removing it I made
the code more concise
